app/services/ValidatingPageBuilder.scala

```diff
@@ -182,21 +190,31 @@ class ValidatingPageBuilder @Inject() (pageBuilder: PageBuilder){
         }
     }
 
-  private def checkForSequencePageReuse(connections: List[PageVertex])(implicit stanzaMap: Map[String, Stanza]): List[GuidanceError] = {
-    val vertexMap = connections.map(pv => (pv.id, pv)).toMap
-    val mainFlow: List[String] = mainFlowPageIds(vertexMap)
-    val sequencePageIds = for{
-      pv <- connections.filterNot(_.flows.isEmpty)              // Sequences
+  private def checkAllFlowsHaveUniqueTerminationPage(vertices: List[PageVertex])(implicit stanzaMap: Map[String, Stanza]): List[GuidanceError] = {
+    val vertexMap = vertices.map(pv => (pv.id, pv)).toMap
+    val mainFlow: List[String] = pageGraph(List(Process.StartStanzaId), vertexMap).map(_.id)
+
+    vertices.filterNot(_.flows.isEmpty) // Sequences
+            .flatMap(_.flows).distinct  // Unique flow ids
+            .filter{id =>               // All those containing pages which dont link to an EndStanza
+              val pages = pageGraph(findPageIds(List(id)), vertexMap, mainFlow)
+              !pages.isEmpty && !pages.exists(_.endPage)
```

Comment:
You could use nonEmpty rather than !...isEmpty.

Reply:
done